tools/mo/openvino/tools/mo/back/ConvolutionNormalizer.py

```diff
@@ -29,22 +33,69 @@ def resolve_convolution_with_group(node: Node, group: int, ir_version: str):
             return
-        new_shape = shape_array([node.output, -1, *weights_shape[2:]])
+
+        num_spatial_dims = len(weights_shape[2:])
+        # Reshape has special_zero=True, if zeros are set then original shapes are copied
+        zeros_to_copy_spatial_dims = np.zeros(num_spatial_dims)
+        new_shape = shape_array([node.output, -1, *zeros_to_copy_spatial_dims])
+        reshape = create_op_node_with_second_input(node.graph, Reshape, new_shape,
+                                                   {'override_output_shape': True})
     elif ir_version == 'V10':
-        # TODO rewrite this transformation to generate a shape-computing sub-graph. Ticket 62076
         I = input_shape[1]
-        new_shape = shape_array([group, node.output // group, I // group, *weights_shape[2:]])
-        assert is_fully_defined(weights_shape[2:]) and is_fully_defined(I) and \
-               np.prod(weights_shape) == np.prod(new_shape), 'Initial weights shape {}, grouped weights shape {}' \
-                                                             ''.format(weights_shape, new_shape)
+        if is_fully_defined(weights_shape[2:]) and is_fully_defined(I):
+            new_shape = shape_array([group, node.output // group, I // group, *weights_shape[2:]])
+            assert np.prod(weights_shape) == np.prod(new_shape), 'Initial weights shape {}, grouped weights shape {}' \
+                                                                 ''.format(weights_shape, new_shape)
+            reshape = create_op_node_with_second_input(node.graph, Reshape, new_shape,
+                                                       {'override_output_shape': True})
+        else:
+            # if weights/or input channel dimension is dynamic need to to compute new_shape in a new subgraph
+            weights_node = node.in_port(1).get_source().node
+            input_node = node.in_port(0).get_source().node
+
+            weights_shape = Shape(node.graph, {'name': node_name + '/ShapeOfWeights'}).create_node()
+            weights_shape.in_port(0).connect(weights_node.out_port(0))
+
+            weights_spatial_shape = create_op_with_const_inputs(node.graph, StridedSlice,
+                                                                port_value_dict={1: int64_array([2]),
+                                                                                 2: int64_array([-1])},
+                                                                op_attrs={'begin_mask': [1],
+                                                                          'end_mask': [0],
+                                                                          'new_axis_mask': [0],
+                                                                          'shrink_axis_mask': [0],
+                                                                          'ellipsis_mask': [0]},
+                                                                input_node=weights_shape)
```

Comment:
nit: you can use `node_to_get_shape_value_of_indices()`. Please see an example in `add_reshape_around_convolution.py` from front/kaldi